src/groupeddataframe/splitapplycombine.jl

```diff
@@ -496,6 +496,16 @@ wrap_table(x::Union{NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}},
                     AbstractVector,
                     AbstractMatrix}) = wrap(x)
 
+wrap_row(x::Any) = wrap(x)
+wrap_row(x::Union{AbstractVecOrMat, DataFrame}) =
+    throw(ArgumentError("return value must not change its kind " *
+                        "(single row or variable number of rows) across groups"))
+function wrap_row(x::NamedTuple)
+    if any(x -> x isa AbstractVector, row)
```

Comment:
```suggestion
function wrap_row(x::NamedTuple)
    if any(v -> v isa AbstractVector, x)
```
Also there should be a first branch (o method) `if x isa NamedTuple{<:Any, <:AbstractVector}`, as returning a named tuple of vectors should throw a different error (apparently the existing code wasn't perfect in that regard).

Reply:
I understand you meant `NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}` (these type definitions are nasty). I agree - moved it to the definition above (the one with `AbstractVecOrMat`).